packages/qwik-city/runtime/src/form-component.tsx

```diff
@@ -105,21 +107,23 @@ export const GetForm = component$<FormProps<undefined, undefined>>(
               params.append(key, value);
             }
           });
-          nav('?' + params.toString(), true).then(() => {
```

Comment:
This is changing when the event is getting dispatched!

Reply:
It does look like so, but if you look at how `nav` (i.e. `goto` in `QwikCityProvider`) in detail, you will find that it is actually not an async function coz it neither `await` anything nor returns a Promise. The `submitcompleted` event is dispatched at the same time as it was originally intended.

Reply:
i see, I would say it's a bug of useNavigate then, i would love to wait for navigation to end

Reply:
I just removed the whole 'qnav' and 'qsettled' event change to simplify the change further. This should be addressed now.